### ISSUE TITLE
Tr1d1um is not forwarding the TraceInformation to downstream services.

### ISSUE THREAD

**Issue by Sachin4403** [xmidt-org/tr1d1um#210]
Hi Team,

we have upgraded all the xmidt components in our dev environment according to  https://github.com/xmidt-org/xmidt/discussions/43, But we are getting new traceId every time when the request is transferred from one service to another service.

Root cause of the problem,

we are not forwarding the context which is having the trace information and at the end, it tries to start the span which will result in a new traceId to downstream services.

Here we are skipping the  context forwarding to the new request,
https://github.com/xmidt-org/tr1d1um/blob/149d3e1f02cd26ab78d5110b0df3e2b85872867f/stat/service.go#L53
here it will start the new span because the context will not have the traceinformation.
https://github.com/open-telemetry/opentelemetry-go-contrib/blob/13d72c9c8dda39280ead2e291cbed8f51a6f65f9/instrumentation/net/http/otelhttp/transport.go#L82

A simpler solution to fix this is we need to forward the context like this
```
r = r.withContext(original request context ) 
```

or

``` 
sc:= trace.SpanFromContext(original request context)
ctx = trace.ContextWithSpan(r.contect(),sc)
r = r.withContext(ctx)
```

but we need to take care of this change in every microservice.

Note: It is working with Manual instrumentation which is https://github.com/xmidt-org/tr1d1um/releases/tag/v0.5.5

cc: @joe94 @schmidtw

**Comment by joe94** [comment]
Sounds like a possible disconnect between the server receiving or generating the context and then passing that to the outgoing context from clients. Looking into this.

**Comment by joe94** [comment]
@Sachin4403, I linked a PR with a fix to use the context with traces in outgoing requests for `/config` and `/stat`.

**Comment by Sachin4403** [comment]
Hi @joe94  any plans to  cover  webhook endpoint?

**Comment by joe94** [comment]
I ran the following request which suggests the scytale<->talaria link is working correctly
```
curl -i -H "Authorization: Basic dXNlcjpwYXNz" "http://localhost:6100/api/v2/device/mac:112233445566/config?names=Device.DeviceInfo.X_CISCO_COM_BootloaderVersion" -H "X-Xmidt-Partner-ID: comcast,nbc,sky"
HTTP/1.1 200 OK
Content-Type: application/json; charset=utf-8
X-Midt-Span-Id: b19a12110313fccf
X-Midt-Trace-Id: 9e3e3c9c0b930cf63c1c63ae31ae24c2
X-Scytale-Build: unkown
X-Scytale-Flavor: mint
X-Scytale-Region: east
X-Scytale-Server: 871f79a9ac29
X-Scytale-Start-Time: 19 May 21 16:44 UTC
X-Talaria-Build: unkown
X-Talaria-Flavor: mint
X-Talaria-Region: east
X-Talaria-Server: 4558c20f646c
X-Talaria-Start-Time: 19 May 21 16:53 UTC
X-Tr1d1um-Build: unkown
X-Tr1d1um-Flavor: mint
X-Tr1d1um-Region: east
X-Tr1d1um-Server: 7fa840f25ff8
X-Tr1d1um-Start-Time: 19 May 21 16:44 UTC
X-Webpa-Transaction-Id: Xj_ACfzE0cxfVouBokcjTA
X-Xmidt-Span: "http://898919c43307:6200/api/v2/device/send","2021-05-19T19:15:21Z","30.3589ms"
X-Xmidt-Span: "http://4558c20f646c:6200/api/v2/device/send","2021-05-19T19:15:21Z","35.6407ms"
Date: Wed, 19 May 2021 19:15:21 GMT
Content-Length: 163

{"parameters":[{"name":"Device.DeviceInfo.X_CISCO_COM_BootloaderVersion","value":"4.2.0.45","dataType":0,"parameterCount":1,"message":"Success"}],"statusCode":200}
```
![Screen Shot 2021-05-19 at 12 16 54 PM](https://user-images.githubusercontent.com/5404231/118872069-09e04100-b89d-11eb-99c8-d8f7a1998f6d.png)

**Comment by joe94** [comment]
@Sachin4403, yes, I'm looking into how to resolve the webhook endpoints.

**Comment by Sachin4403** [comment]
To resolve webhook endpoint we need to do the same here

Already context will have the span information

https://github.com/xmidt-org/argus/blob/main/chrysom/client.go#L262

**Comment by joe94** [comment]
Yep, I'm testing that 👍

**Comment by Sachin4403** [comment]
After tr1d1um fix I will try to run E2E and will Inform of there were any issues

**Comment by joe94** [comment]
@Sachin4403, can you run your test from a commit or do you need a release?

**Comment by Sachin4403** [comment]
@joe94  we need release of it. You can release a minor version of it

**Comment by Sachin4403** [comment]
Thanks @joe94  for releasing  v0.5.7 with this fix.

**Comment by joe94** [comment]
@Sachin4403 give `v0.5.7` a try. I'm reopening the issue until you can verify things are working as expected.

**Comment by Sachin4403** [comment]
Sure @joe94  Once E2E is done will inform here.

**Comment by Sachin4403** [comment]
Hello @joe94,

I had checked it E2E and it is working as expected.

**Comment by joe94** [comment]
Hey @Sachin4403, I released `v0.5.8` which includes this fix + the latest official version of `ancla` and `argus`. I recommend upgrading to this version when you can 👍
